planner/scan.go

```diff
@@ -60,17 +63,84 @@ func (n *scanNode) Kind() string {
 
 func (n *scanNode) Init() error {
 	// init the fetcher
-	if err := n.fetcher.Init(&n.desc, n.fields, n.reverse, n.showDeleted); err != nil {
+	if err := n.fetcher.Init(&n.desc, n.fields, n.filter, n.slct.DocumentMapping, n.reverse, n.showDeleted); err != nil {
 		return err
 	}
 	return n.initScan()
 }
 
 func (n *scanNode) initCollection(desc client.CollectionDescription) error {
 	n.desc = desc
+	n.initFields(n.slct.Fields)
+	if n.slct.OrderBy != nil {
+		for _, cond := range n.slct.OrderBy.Conditions {
+			for _, index := range cond.FieldIndexes {
+				name, _ := n.docMapper.documentMapping.TryToFindNameFromIndex(index)
+				n.tryAddField(name)
+			}
+		}
+	}
 	return nil
 }
 
+func (n *scanNode) initFields(fields []mapper.Requestable) {
+	for _, r := range fields {
+		// add all the possible base level fields the fetcher is responsible
+		// for, including those that are needed by higher level aggregates
+		// or grouping alls, which them selves might have further dependants
+		switch requestable := r.(type) {
+		// field is simple as its just a base level field
+		case *mapper.Field:
+			n.tryAddField(requestable.GetName())
+		// select might have its own select fields and filters fields
+		// @todo: Might need to check that these are sub selects
+		// are _groups
+		case *mapper.Select:
+			if requestable.Targetable.Filter != nil {
+				fieldDescs := parser.ParseFilterFieldsForDescription(
+					requestable.Targetable.Filter.ExternalConditions,
+					n.desc.Schema,
+				)
+				for _, fd := range fieldDescs {
+					n.tryAddField(fd.Name)
+				}
+			}
+			n.tryAddField(requestable.Field.Name + "_id") // foreign key for type joins
+			n.initFields(requestable.Fields)
+		// aggregate might have its own target fields and filter fields
+		case *mapper.Aggregate:
```

Comment:
question: Similar to the `*mapper.Select` case - are you really sure this is possible here? And if so, how?

Reply:
yupp

Reply:
Same as other comment, please expand

Reply:
note: Cant (at least at the moment) remove the filter loop like we did with the select case, as its a little more complicated with the potential group stuff. Couldnt potentially update some of the mapper to handle things, but it gets into the weeds even more than we already have xD. Let me know, happy to dive into it and make further changes so we can remove the filter loop here.